### PR TITLE
Nest transaction entities

### PR DESCRIPTION
**Pull request by cdupuis**
Signed-off-by: Christian Dupuis <cd@atomist.com>

### CODE REVIEW

File: handlers_test.go [hunk 1]
```diff
@@ -62,34 +62,35 @@ func TestProcessCommit(t *testing.T) {
...
-				commitEntity := s.Index(1).Interface().(GitCommitEntity)
+				commitSignatureEntity := s.Index(0).Interface().(GitCommitSignatureEntity)
+				commitEntity := commitSignatureEntity.Commit
 				if !reflect.DeepEqual(commitEntity, GitCommitEntity{
 					Entity: skill.Entity{
 						EntityType: edn.Keyword("git/commit"),
 						Entity:     "$commit",
 					},
-					Repo: "$repo",
+					Repo: assertRepoEntity,
```
Comment (errordeveloper) [inline handlers_test.go:84]: I think not having the dollar-sign notation here make it much easier to read!

File: handlers.go [hunk 2]
```diff
@@ -103,18 +103,18 @@ func transactCommitSignature(ctx context.Context, req skill.RequestContext, comm
...
-		Sha:    commit.Sha,
-		Repo:   "$repo",
-		Url:    commit.Repo.Org.Url,
-	}, GitCommitSignatureEntity{
-		Entity:    transaction.MakeEntity("git.commit/signature"),
-		Commit:    "$commit",
+	transaction.AddEntities(GitCommitSignatureEntity{
+		Entity: transaction.MakeEntity("git.commit/signature"),
+		Commit: GitCommitEntity{
+			Entity: transaction.MakeEntity("git/commit", "$commit"),
```
Comment (errordeveloper) [inline handlers.go:109]: Can this be done without dolar-sing anywhere? I am just wondering it's something we could leave out for more advanced use-cases?

**Review by errordeveloper — APPROVED** [review]
LGTM, just a few thoughts in comments...

**Comment by cdupuis** [comment]
> LGTM, just a few thoughts in comments...

How is the latest?

**Comment by errordeveloper** [comment]
> > LGTM, just a few thoughts in comments...
> 
> How is the latest?

Nice!

**Pull request merged**